dist/src/main/java/io/camunda/zeebe/gateway/ActorSchedulerConfiguration.java

```diff
@@ -10,23 +10,28 @@
 import io.camunda.zeebe.gateway.impl.configuration.GatewayCfg;
 import io.camunda.zeebe.scheduler.ActorScheduler;
 import io.camunda.zeebe.shared.ActorClockConfiguration;
+import io.camunda.zeebe.shared.IdleStrategyConfig;
 import io.camunda.zeebe.util.VisibleForTesting;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
-import org.springframework.stereotype.Component;
+import org.springframework.context.annotation.Configuration;
 
-@Component
+@Configuration(proxyBeanMethods = false)
```

Comment:
❓ Not sure what this means, could you explain?

Reply:
`@Component` is the top level generic Spring annotation to denote components (lol) that are candidate for auto-detection by Spring - meaning they can be auto wired, post processed, etc., all the Spring magic.

`@Configuration` is a specific type of component which serves bean and bean definitions (and a bit more, but generally that's the gist of it). By default, all beans (i.e. all methods annotated with `@Bean`) instantiated and managed by Spring are actually Java proxies produced via [cglib](https://github.com/cglib/cglib). This can be useful for instrumentation purposes, [aspect weaving](https://eclipse.dev/aspectj/), etc, but it also has some downsides (poor tooling - close to no IDE/debugger support, unreadable stack traces, etc). 

Since we don't make use of any of these things, I think for us it makes sense to turn it off where possible.

Reply:
Sorry @npepinpe I'm totally a spring noob. Thanks for the clarification.